dora/core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -709,34 +712,51 @@ public boolean rename(Path src, Path dst) throws IOException {
       return false;
     } catch (InvalidArgumentRuntimeException e) {
       throw new IllegalArgumentException(e);
-    } catch (AlluxioRuntimeException e) {
-      throw toHdfsIOException(e);
-    } catch (AlluxioException e) {
-      ensureExists(srcPath);
-      URIStatus dstStatus;
-      try {
-        dstStatus = mFileSystem.getStatus(dstPath);
-      } catch (IOException | AlluxioException e2) {
-        LOG.warn("rename failed: {}", e.toString());
-        return false;
-      }
-      // If the destination is an existing folder, try to move the src into the folder
-      if (dstStatus != null && dstStatus.isFolder()) {
-        dstPath = dstPath.joinUnsafe(srcPath.getName());
-      } else {
-        LOG.warn("rename failed: {}", e.toString());
-        return false;
+    } catch (AlluxioRuntimeException | AlluxioException e) {
+      Exception exToLog = e;
+      if (e instanceof  AlluxioRuntimeException) {
+        exToLog = toHdfsIOException((AlluxioRuntimeException) e);
       }
       try {
-        mFileSystem.rename(srcPath, dstPath);
-      } catch (IOException | AlluxioException e2) {
-        LOG.error("Failed to rename {} to {}", src, dst, e2);
+        boolean res = tryMoveIntoDirectory(src, dst, srcPath, dstPath);
+        if (!res) {
+          LOG.warn("Failed to rename file and dst {} is not a directory", dst, exToLog);
+        }
+      } catch (IOException | AlluxioException | AlluxioRuntimeException e2) {
+        LOG.error("Failed to rename file {} and failed to move into dst directory {}",
```

Comment:
can we separate this log into two logs? like:
```
LOG.error("Failed to rename {} to {}, exception:{}", src, 1st dst, 1st ex);
LOG.error("Failed to rename {} to {},  exception:{}", src, 2nd dst, 2nd ex);
```

Reply:
I feel 2 messages is hard to read and you may even see irrelevant messages in the middle, so what about
```
LOG.error("Failed to rename src {} to path {} with exception {} and failed to move src to dst as a folder with exception", src, dst, 1st ex, 2nd ex);
```
So 1st ex will print a `ex.toString()` which is `Exception class name: message` and the 2nd ex has full stacktrace. I hope the 2nd stacktrace gives you enough info that you don't need the stack trace for 1st.